pkg/virt-api/rest/subresource.go

```diff
@@ -1545,3 +1550,239 @@ func (app *SubresourceAPIApp) RemoveMemoryDumpVMRequestHandler(request *restful.
+func (app *SubresourceAPIApp) VMIAddInterfaceRequestHandler(request *restful.Request, response *restful.Response) {
+	app.addInterfaceRequestHandler(request, response, false)
+}
+
+func (app *SubresourceAPIApp) addInterfaceRequestHandler(request *restful.Request, response *restful.Response, persisted bool) {
+	name := request.PathParameter("name")
+	namespace := request.PathParameter("namespace")
+
+	if !app.clusterConfig.HotplugNetworkInterfacesEnabled() {
+		writeError(errors.NewBadRequest("Unable to Add Interface because HotplugNICs feature gate is not enabled."), response)
+		return
+	}
+
+	opts := &v1.AddInterfaceOptions{}
+	if request.Request.Body != nil {
+		defer request.Request.Body.Close()
+		err := yaml.NewYAMLOrJSONDecoder(request.Request.Body, 1024).Decode(opts)
+		switch err {
+		case io.EOF, nil:
+			break
+		default:
+			writeError(errors.NewBadRequest(fmt.Sprintf("Can not unmarshal Request body to struct, error: %s", err)), response)
+			return
+		}
+	} else {
+		writeError(errors.NewBadRequest("`networkName` and `interfaceName` are expected"), response)
+		return
+	}
+
+	if opts.NetworkName == "" {
+		writeError(errors.NewBadRequest("AddInterfaceOptions requires `networkName` to be set"), response)
+		return
+	} else if opts.InterfaceName == "" {
+		writeError(errors.NewBadRequest("AddInterfaceOptions requires `interfaceName` to be set"), response)
+		return
+	}
+
+	interfaceRequest := v1.VirtualMachineInterfaceRequest{
+		AddInterfaceOptions: opts,
+	}
+
+	if persisted {
+		// set as a request on the VM to both make permanent and hotplug.
+		if err := app.vmInterfacePatchStatus(name, namespace, &interfaceRequest); err != nil {
+			writeError(err, response)
+			return
+		}
+	} else {
+		// inject into VMI when ephemeral
+		if err := app.vmiInterfacePatch(name, namespace, &interfaceRequest); err != nil {
+			writeError(err, response)
+			return
+		}
+	}
+
+	response.WriteHeader(http.StatusAccepted)
+}
+
+func (app *SubresourceAPIApp) vmiInterfacePatch(vmName string, namespace string, interfaceRequest *v1.VirtualMachineInterfaceRequest) *errors.StatusError {
+	vmi, statErr := app.FetchVirtualMachineInstance(namespace, vmName)
+	if statErr != nil {
+		return statErr
+	}
+
+	if !vmi.IsRunning() {
+		return errors.NewConflict(v1.Resource("virtualmachineinstance"), vmName, fmt.Errorf("VMI is not running"))
+	}
+
+	patch, err := generateVMIInterfaceRequestPatch(vmi, interfaceRequest)
+	if err != nil {
+		return errors.NewConflict(v1.Resource("virtualmachineinstance"), vmName, err)
+	}
+
+	log.Log.Object(vmi).V(4).Infof("Patching VMI: %s", patch)
+	if _, err := app.virtCli.VirtualMachineInstance(vmi.Namespace).Patch(context.Background(), vmi.Name, types.JSONPatchType, []byte(patch), &k8smetav1.PatchOptions{}); err != nil {
+		log.Log.Object(vmi).V(1).Errorf("unable to patch vmi: %v", err)
+		if errors.IsInvalid(err) {
+			if statErr, ok := err.(*errors.StatusError); ok {
```

Comment:
I think you are suppose to use [errors.As](https://pkg.go.dev/errors#example-As)  instead of this type casting thing.